src/main/java/org/zalando/fahrschein/NakadiReaderFactory.java

```diff
@@ -0,0 +1,36 @@
+package org.zalando.fahrschein;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.zalando.fahrschein.domain.Subscription;
+import org.zalando.fahrschein.metrics.MetricsCollector;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.Optional;
+
+public class NakadiReaderFactory {
```

Comment:
Could be package private or do you see a usecase of accessing this from outside the library?